contracts/DeployDebtToken.sol

```diff
@@ -0,0 +1,52 @@
+import './DebtToken.sol';
+pragma solidity ^0.4.15;
+contract DeployDebtToken is Ownable{
+
+    address public owner;
+    address public dayTokenAddress;
+    uint public dayTokenFees; //DAY tokens to be paid for deploying custom DAY contract
+    ERC20 dayToken;
+
+    event FeeUpdated(uint _fee, uint _time);
+    event DebtTokenCreated(address  _creator, address _debtTokenAddress, uint256 _time);
+
+    function DeployDebtToken(address _dayTokenAddress, uint _dayTokenFees){
+        dayTokenAddress = _dayTokenAddress;
+        dayTokenFees = _dayTokenFees;
+        owner = msg.sender;
+        dayToken = ERC20(dayTokenAddress);
+    }
+
+    function updateDayTokenFees(uint _dayTokenFees) onlyOwner public {
+        dayTokenFees = _dayTokenFees;
+        FeeUpdated(dayTokenFees, now);
+    }
+
+    function createDebtToken(string _tokenName,
+        string _tokenSymbol,
+        uint256 _initialAmount,
+        uint256 _exchangeRate,
+        uint256 _decimalUnits,
+        uint256 _dayLength,
+        uint256 _loanTerm,
+        uint256 _loanCycle,
+        uint256 _interestRate,
+        address _debtOwner){
+
+        address user = msg.sender;
+
+        if(dayToken.transferFrom(user, this, dayTokenFees)){
+            DebtToken newDebtToken = new DebtToken(_tokenName, _tokenSymbol, _initialAmount, _exchangeRate,
+                _decimalUnits, _dayLength, _loanTerm, _loanCycle,
+                _interestRate, _debtOwner);
+            DebtTokenCreated(user, address(newDebtToken), now);
+        }
+
+    }
+
+    // to collect all fees paid till now
+    function fetchDayTokens() onlyOwner public {
+        dayToken.transfer(owner, dayToken.balanceOf(this));
+    }
+
```

Comment:
can we add the fallback function, just to ensure that ether are not transferred to the address?

Reply:
by default fallback is not implemented, which means that any tx with ETH will be reverted

Reply:
Please add tests for deployment code.